### PR TITLE
Set up go.mod before every test

### PR DESCRIPTION
**Pull request by Joakker**
With this PR, hopefully most of the tests for the Go runtime would be fixed, as the errors from the tool seem to be because the testing isn't being done on a go module.

At the request of @mitar in #3174

### CODE REVIEW

File: runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java [hunk 1]
```diff
@@ -80,10 +80,36 @@ private static void cacheGoRuntime(File tmpPackageDir) throws Exception {
...
+        if (exit != 0) {
+            throw new Exception("Non-zero exit while setting up go.mod:" + sucker.toString());
+        }
+    }
+
 	public void testSetUp() throws Exception {
 		eraseParserTempDir();
 		super.testSetUp();
 		parserTempDir = new File(getTempTestDir(), "parser");
+        setupGoMod();
```
Comment (mitar) [inline runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java:112]: Indent.

File: runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java [hunk 2]
```diff
@@ -80,10 +80,36 @@ private static void cacheGoRuntime(File tmpPackageDir) throws Exception {
 		}
 	}
 
+    private static void setupGoMod() throws Exception{
```
Comment (mitar) [inline runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java:83]: Are you sure we have to do this every time and not only once inside `cacheGoRuntime`?

Comment (Joakker) [reply runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java:83]: True, I'll refactor it

File: runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java (hunk 1 above)
Comment (Joakker) [reply runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseGoTest.java:112]: Oop, my bad

**Comment by Joakker** [comment]
Ok, ok, ok, now most checks are good, but the last one fails. That's progress 😃 !

**Comment by Joakker** [comment]
Closing this, as I'll be rewritting the tests from scratch. That'll let me keep my sanity

**Pull request closed without merging**